### PR TITLE
Fix VAO

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: libraries/gpu/src/gpu/GLBackendTransform.cpp [hunk 1]
```diff
@@ -224,11 +218,14 @@ void GLBackend::updateTransform(const Batch& batch) {
             glEnableVertexAttribArray(gpu::Stream::DRAW_CALL_INFO); // Make sure attrib array is enabled
             glVertexAttribIFormat(gpu::Stream::DRAW_CALL_INFO, 2, GL_UNSIGNED_SHORT, 0);
             glVertexAttribBinding(gpu::Stream::DRAW_CALL_INFO, gpu::Stream::DRAW_CALL_INFO);
-            glVertexBindingDivisor(gpu::Stream::DRAW_CALL_INFO, 1);
             _transform._enabledDrawcallInfoBuffer = true;
         }
-        glBindVertexBuffer(gpu::Stream::DRAW_CALL_INFO, _transform._drawCallInfoBuffer, (GLintptr)_transform._drawCallInfoOffsets[batch._currentNamedCall], 0);
-
+        // NOTE: A stride of zero in BindVertexBuffer signifies that all elements are sourced from the same location,
```
Comment (zzmp) [inline libraries/gpu/src/gpu/GLBackendTransform.cpp:224]: See the issues section of [the spec](https://www.opengl.org/registry/specs/ARB/vertex_attrib_binding.txt).

File: libraries/gpu/src/gpu/GLBackendTransform.cpp [hunk 2]
```diff
@@ -224,11 +218,14 @@ void GLBackend::updateTransform(const Batch& batch) {
...
+        //       so we must provide a stride.
+        //       This is in contrast to VertexAttrib*Pointer, where a zero signifies tightly-packed elements.
+        glBindVertexBuffer(gpu::Stream::DRAW_CALL_INFO, _transform._drawCallInfoBuffer, (GLintptr)_transform._drawCallInfoOffsets[batch._currentNamedCall], 2 * sizeof(GLushort));
+        // NOTE: Instance divisor is per-binding, so it must be specified after BindVertexBuffer.
+        glVertexAttribDivisor(gpu::Stream::DRAW_CALL_INFO, 1);
```
Comment (samcake) [inline libraries/gpu/src/gpu/GLBackendTransform.cpp]: can we do the divisor in the if block above ?

Comment (samcake) [inline libraries/gpu/src/gpu/GLBackendTransform.cpp]: it has to be a glVertexBindingDivisor

File: libraries/gpu/src/gpu/GLBackendInput.cpp [hunk 3]
```diff
@@ -23,28 +23,24 @@ GLBackend::GLInputFormat:: ~GLInputFormat() {
...
+    if(GLBackend::syncGPUObject(*format)) {
         if (format != _input._format) {
             _input._format = format;
+            _input._invalidFormat = true;
             if (format) {
-                if (_input._formatKey != format->getKey()) {
-                    _input._formatKey = format->getKey();
-                    _input._invalidFormat = true;
-                }
+                _input._formatKey = format->getKey();
```
Comment (samcake) [inline libraries/gpu/src/gpu/GLBackendInput.cpp]: no no nonononononono

File: libraries/gpu/src/gpu/GLBackendTransform.cpp [hunk 4]
```diff
@@ -227,8 +221,10 @@ void GLBackend::updateTransform(const Batch& batch) {
             glVertexBindingDivisor(gpu::Stream::DRAW_CALL_INFO, 1);
             _transform._enabledDrawcallInfoBuffer = true;
         }
-        glBindVertexBuffer(gpu::Stream::DRAW_CALL_INFO, _transform._drawCallInfoBuffer, (GLintptr)_transform._drawCallInfoOffsets[batch._currentNamedCall], 0);
-
+        // NOTE: A stride of zero in BindVertexBuffer signifies that all elements are sourced from the same location,
+        //       so we must provide a stride.
+        //       This is in contrast to VertexAttrib*Pointer, where a zero signifies tightly-packed elements.
+        glBindVertexBuffer(gpu::Stream::DRAW_CALL_INFO, _transform._drawCallInfoBuffer, (GLintptr)_transform._drawCallInfoOffsets[batch._currentNamedCall], 2 * sizeof(GLushort));
```
Comment (samcake) [inline libraries/gpu/src/gpu/GLBackendTransform.cpp:227]: brilliant

**Pull request merged**